drivers/storage/port/scsiport/scsi.c

```diff
@@ -653,12 +653,15 @@ SpiProcessCompletedRequest(
     PSCSI_PORT_LUN_EXTENSION LunExtension;
     LONG Result;
     PIRP Irp;
-    //ULONG SequenceNumber;
+    ULONG transferLen = 0;
+    PSCSI_SG_ADDRESS pListIter = SrbInfo->ScatterGather;
```

Comment:
You may move those two variables, + the isWrite below, into the only code sub-block (the `if(SrbInfo->BaseOfMapRegister)`) in which they are used.